tst/EnergyPlus/unit/SolarShading.unit.cc

```diff
@@ -6204,3 +6204,285 @@ TEST_F(EnergyPlusFixture, SolarShadingTest_GetShadowingInputTest6)
+        // From 2 to 7
+        TestCase{Line{Point{greater_x, below_y}, Point{center_x, greater_y}}, true, Line{Point{maxX, 4.2}, Point{6.875, maxY}}},
+        // From 2 to 5
+        TestCase{Line{Point{greater_x, below_y}, Point{greater_x, center_y}}, false},
+        // From 2 to 8
+        TestCase{Line{Point{greater_x, below_y}, Point{greater_x, greater_y}}, false},
+        // From 5 to 0
+        TestCase{Line{Point{greater_x, center_y}, Point{below_x, below_y}}, true, Line{Point{maxX, 3.8}, Point{5.714285714285714, minY}}},
+        // From 5 to 3
+        TestCase{Line{Point{greater_x, center_y}, Point{below_x, center_y}}, true, Line{Point{maxX, center_y}, Point{minX, center_y}}},
+        // From 5 to 6
+        TestCase{Line{Point{greater_x, center_y}, Point{below_x, greater_y}}, true, Line{Point{maxX, 5.4}, Point{6.666666666666667, maxY}}},
+        // From 5 to 1
+        TestCase{
+            Line{Point{greater_x, center_y}, Point{center_x, below_y}}, true, Line{Point{maxX, 3.0999999999999996}, Point{7.857142857142858, minY}}},
+        // From 5 to 4
+        TestCase{Line{Point{greater_x, center_y}, Point{center_x, center_y}}, true, Line{Point{maxX, center_y}, Point{center_x, center_y}}},
+        // From 5 to 7
+        TestCase{Line{Point{greater_x, center_y}, Point{center_x, greater_y}}, false},
+        // From 5 to 2
+        TestCase{Line{Point{greater_x, center_y}, Point{greater_x, below_y}}, false},
+        // From 5 to 8
+        TestCase{Line{Point{greater_x, center_y}, Point{greater_x, greater_y}}, false},
+        // From 8 to 0
+        TestCase{Line{Point{greater_x, greater_y}, Point{below_x, below_y}}, true, Line{Point{6.25, maxY}, Point{2.5, minY}}},
+        // From 8 to 3
+        TestCase{Line{Point{greater_x, greater_y}, Point{below_x, center_y}}, true, Line{Point{3.333333333333334, maxY}, Point{minX, 5.4}}},
+        // From 8 to 6
+        TestCase{Line{Point{greater_x, greater_y}, Point{below_x, greater_y}}, false},
+        // From 8 to 1
+        TestCase{Line{Point{greater_x, greater_y}, Point{center_x, below_y}}, true, Line{Point{maxX, 5.8}, Point{6.25, minY}}},
+        // From 8 to 4
+        TestCase{Line{Point{greater_x, greater_y}, Point{center_x, center_y}}, true, Line{Point{6.666666666666667, maxY}, Point{center_x, center_y}}},
+        // From 8 to 7
+        TestCase{Line{Point{greater_x, greater_y}, Point{center_x, greater_y}}, false},
+        // From 8 to 2
+        TestCase{Line{Point{greater_x, greater_y}, Point{greater_x, below_y}}, false},
+        // From 8 to 5
+        TestCase{Line{Point{greater_x, greater_y}, Point{greater_x, center_y}}, false},
+    }};
+
+    size_t i = 0;
+    for (const auto &t : test_cases) {
+        ++i;
+        std::string const msg =
+            fmt::format("test_case {}: From ({}, {}) to ({}, {})", i, t.line_ori.p0.x, t.line_ori.p0.y, t.line_ori.p1.x, t.line_ori.p1.y);
+        SCOPED_TRACE(msg);
```

Comment:
Does this always emit the current position?  I will see it locally if so, just asking to remind myself to look I guess.

Reply:
OK, that's great. I thought it was just emitting info by calling the macro itself.  I now see it is just registering the extra data to report in case of a failure.  This is great, thanks.